modules/ducktests/src/main/java/org/apache/ignite/internal/ducktest/tests/cellular_affinity_test/CellularTxStreamer.java

```diff
@@ -58,26 +63,30 @@
 
         Affinity<Integer> aff = ignite.affinity(cacheName);
 
-        while (!terminated()) {
-            key++;
+        List<Integer> cellKeys = new ArrayList<>();
+
+        int candidate = 0;
 
-            Collection<ClusterNode> nodes = aff.mapKeyToPrimaryAndBackups(key);
+        while (cellKeys.size() < 100) {
+            Collection<ClusterNode> nodes = aff.mapKeyToPrimaryAndBackups(++candidate);
 
             Map<Object, Long> stat = nodes.stream().collect(
```

Comment:
Are you sure that creating the same map on each iteration is a good idea?

Reply:
each key has it's own map

Reply:
Sorry, collections difers from each other on each iteration